applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.cpp

```diff
@@ -1132,142 +1135,552 @@ void PrestressMembraneElement::CalculateMembraneElasticityTensor(
+
+    if (mConstitutiveLawVector.size() == 0)
+    {
+        mConstitutiveLawVector.resize(integration_points.size());
+
+        for (unsigned int i = 0; i < mConstitutiveLawVector.size(); i++)
+        {
+            mConstitutiveLawVector[i] = GetProperties()[CONSTITUTIVE_LAW]->Clone();
+
+            mConstitutiveLawVector[i]->InitializeMaterial(GetProperties(), GetGeometry(), row(GetGeometry().ShapeFunctionsValues(), i));
+        }
+    }
+}
+void PrestressMembraneElement::InitializeNonLinearIteration(){
+    
+}
+void PrestressMembraneElement::UpdatePrestress(unsigned int PointNumber){
+   
+    //-------------------------------------------
+    //--1--Computation different CoSys needed for the prestress transformation
+    //-------------------------------------------
+
+    // Compute actual cosy and metric
+    array_1d<double, 3> g1;
+    array_1d<double, 3> g2;
+    array_1d<double, 3> g3;
+    array_1d<double, 3> gab;
     
-    // finally, calculating the Transformation Matrix
-    Tm(0, 0) = eG11*eG11;
-    Tm(0, 1) = eG12*eG12;
-    Tm(0, 2) = 2.0*eG11*eG12;
+    const GeometryType::ShapeFunctionsGradientsType& DN_DeContainer = GetGeometry().ShapeFunctionsLocalGradients();
+    Matrix DN_De = DN_DeContainer[PointNumber];
+    CalculateMetricDeformed(PointNumber, DN_De, gab, g1, g2);
+
+    // compute the out-of-plane direction and normalize it
+    CrossProduct(g3,g1,g2);
+    g3 /= norm_2(g3);
+
+    // computation of the out-of-plane direction in the reference configuration
+    array_1d<double, 3> G3;
+    CrossProduct(G3,mG1[PointNumber],mG2[PointNumber]);
+    G3 /= norm_2(G3);
+
+    //Compute cartesian basis in total reference configuration
+    // E1_tot = mG1/|mG1|, E3_tot = mG3Initial
+    array_1d<double, 3> E1_tot,E2_tot, E3_tot;
+    E1_tot = mG1Initial[PointNumber]/norm_2(mG1Initial[PointNumber]);
+    E3_tot = mG3Initial[PointNumber];
+    CrossProduct(E2_tot,E3_tot,E1_tot);
+    E2_tot /= norm_2(E2_tot);
+
+    //Compute cartesian basis in (updated) reference configuration
+    // E1 = mG1/|mG1|, E3 = mG3Initial
+    array_1d<double, 3> E1, E2, E3;
+    E1 = mG1[PointNumber]/norm_2(mG1[PointNumber]);
+    E3 = G3;
+    CrossProduct(E2,E3,E1);
+    E2 /= norm_2(E2);
+
+    // Compute contravariant basis in total Reference configuration
+    // -->Compute the metric in total reference configuration
+    array_1d<double, 3> metric_reference_tot;
+    metric_reference_tot(0)=inner_prod(mG1Initial[PointNumber],mG1Initial[PointNumber]);
+    metric_reference_tot(1)=inner_prod(mG2Initial[PointNumber],mG2Initial[PointNumber]);
+    metric_reference_tot(2)=inner_prod(mG2Initial[PointNumber],mG1Initial[PointNumber]);
+
+    // -->Invert metric in (total) reference configuration
+    double det_metric_tot = metric_reference_tot(0)*metric_reference_tot(1)-metric_reference_tot(2)*metric_reference_tot(2);
+    array_1d<double, 3> inv_metric_tot;
+    inv_metric_tot(0) = 1.0/det_metric_tot * metric_reference_tot(1);
```

Comment:
here you devide by sth that could be zero.
This would give a segfault and you don't know why.

You could use a check, sth like this:
`KRATOS_ERROR_IF(det_metric_tot < std::numeric_limits<double>::epsilon()) << "devision by zero!" << std::endl;`

Reply:
maybe the check could be done only in debug not to have an effect on performance

Reply:
done